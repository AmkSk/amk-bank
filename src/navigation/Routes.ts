export enum Routes {
  IntroScreen = 'IntroScreen',
  WelcomeScreen = 'WelcomeScreen',
  LoginScreen = 'LoginScreen',
  TosModal = 'TosModal',
  PrivacyPolicyModal = 'PrivacyPolicyModal',
  // Onboarding
  OnboardingCreateAccountScreen = 'OnboardingCreateAccountScreen',
  OnboardingEmailScreen = 'OnboardingEmailScreen',
  OnboardingPersonalInfoScreen = 'OnboardingPersonalInfoScreen',
}
