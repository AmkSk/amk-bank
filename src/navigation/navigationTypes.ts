export enum Routes {
  IntroScreen = 'IntroScreen',
  WelcomeScreen = 'WelcomeScreen',
  LoginScreen = 'LoginScreen',
  TosModal = 'TosModal',
  PrivacyPolicyModal = 'PrivacyPolicyModal',
  // Onboarding
  OnboardingCreateAccountScreen = 'OnboardingCreateAccountScreen',
  OnboardingEmailScreen = 'OnboardingEmailScreen',
  OnboardingPersonalInfoScreen = 'OnboardingPersonalInfoScreen',
  OnboardingCountryScreen = 'OnboardingCountryScreen',
  // Authenticated Screens
  TabNavigator = 'TabNavigator',
  DashboardScreen = 'DashboardScreen',
  AnalyticsScreen = 'AnalyticsScreen',
  SettingsScreen = 'SettingsScreen',
  SupportScreen = 'SupportScreen',
}

export type RootStackParamList = {
  [Routes.LoginScreen]: undefined
  [Routes.WelcomeScreen]: undefined
  [Routes.IntroScreen]: undefined
  [Routes.TosModal]: undefined
  [Routes.PrivacyPolicyModal]: undefined
  // Onboarding
  [Routes.OnboardingCreateAccountScreen]: undefined
  [Routes.OnboardingEmailScreen]: undefined
  [Routes.OnboardingPersonalInfoScreen]: undefined
  [Routes.OnboardingCountryScreen]: undefined
  // Authenticated Screens in BottomTab
  [Routes.TabNavigator]: undefined
}

export type TabParamList = {
  // Authenticated Screens
  [Routes.DashboardScreen]: undefined
  [Routes.AnalyticsScreen]: undefined
  [Routes.SettingsScreen]: undefined
  [Routes.SupportScreen]: undefined
}
