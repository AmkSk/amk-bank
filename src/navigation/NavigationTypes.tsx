import { Routes } from './Routes'

export type RootStackParamList = {
  [Routes.LoginScreen]: undefined
  [Routes.WelcomeScreen]: undefined
  [Routes.IntroScreen]: undefined
  [Routes.TosModal]: undefined
  [Routes.PrivacyPolicyModal]: undefined
  // Onboarding
  [Routes.OnboardingCreateAccountScreen]: undefined
  [Routes.OnboardingEmailScreen]: undefined
}
