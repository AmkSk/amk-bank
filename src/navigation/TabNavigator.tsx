import { createBottomTabNavigator } from '@react-navigation/bottom-tabs'
import { Routes, TabParamList } from './navigationTypes'
import { DashboardScreen } from '../screens/user/DashboardScreen'
import { SettingsScreen } from '../screens/user/SettingsScreen'
import { SupportScreen } from '../screens/user/SupportScreen'
import { AnalyticsScreen } from '../screens/user/AnalyticsScreen'
import Icons from '@expo/vector-icons/Ionicons'
import { Platform, StyleSheet } from 'react-native'
import { MD3Theme, useTheme } from 'react-native-paper'

const BottomTab = createBottomTabNavigator<TabParamList>()

export function TabNavigator() {
  const theme = useTheme()
  const styles = createStyleSheet(theme)

  return (
    <BottomTab.Navigator
      screenOptions={{
        tabBarShowLabel: false,
        tabBarStyle: styles.tabBarBackground,
      }}
    >
      <BottomTab.Screen
        name={Routes.DashboardScreen}
        component={DashboardScreen}
        options={{
          headerShown: false,
          tabBarIcon: (params) => <TabBarIcon color={params.color} name={params.focused ? 'home' : 'home-outline'} />,
        }}
      />
      <BottomTab.Screen
        name={Routes.AnalyticsScreen}
        component={AnalyticsScreen}
        options={{
          tabBarIcon: (params) => (
            <TabBarIcon color={params.color} name={params.focused ? 'pie-chart' : 'pie-chart-outline'} />
          ),
        }}
      />
      <BottomTab.Screen
        name={Routes.SupportScreen}
        component={SupportScreen}
        options={{
          tabBarIcon: (params) => (
            <TabBarIcon color={params.color} name={params.focused ? 'chatbubble' : 'chatbubble-outline'} />
          ),
        }}
      />
      <BottomTab.Screen
        name={Routes.SettingsScreen}
        component={SettingsScreen}
        options={{
          tabBarIcon: (params) => (
            <TabBarIcon color={params.color} name={params.focused ? 'settings' : 'settings-outline'} />
          ),
        }}
      />
    </BottomTab.Navigator>
  )
}

interface TabBarIconProps {
  color: string
  name: any
}

function TabBarIcon({ color, name }: TabBarIconProps) {
  return <Icons size={16} name={name} color={color} />
}

const createStyleSheet = (theme: MD3Theme) =>
  StyleSheet.create({
    tabBarBackground: {
      borderRadius: 12,
      margin: Platform.OS === 'ios' ? 0 : 16,
      backgroundColor: theme.colors.background,
    },
  })
