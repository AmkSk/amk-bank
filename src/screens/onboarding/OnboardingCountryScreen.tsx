import { NativeStackScreenProps } from 'react-native-screens/native-stack'
import { RootStackParamList, Routes } from '../../navigation/navigationTypes'
import { ScreenTemplate } from '../ScreenTemplate'
import { Button, Dialog, Portal, Text } from 'react-native-paper'
import { Strings } from '../../i18n/strings'
import { View } from 'react-native'
import { useEffect, useState } from 'react'
import { CommonStyles } from '../../themes/CommonStyles'
import { Country } from '../../data/types'
import { PaperSelect } from 'react-native-paper-select'
import { SelectedItem } from 'react-native-paper-select/lib/typescript/interface/paperSelect.interface'
import { useOnboardingStore } from '../../stores/onboardingStore'
import { CommonActions } from '@react-navigation/native'
import { useGetCountries } from './hooks/useGetCountries'
import { LoadingModal } from '../../components/LoadingModal'
import { useCreateUser } from './hooks/useCreateUser'

export function OnboardingCountryScreen({
  navigation,
}: NativeStackScreenProps<RootStackParamList, Routes.OnboardingCountryScreen>) {
  const [isButtonEnabled, setIsButtonEnabled] = useState(false)
  const [isSuccessDialogShown, showSuccessDialog] = useState(false)

  const selectedCountry = useOnboardingStore((state) => state.countryOfResidence)
  const setSelectedCountry = useOnboardingStore((state) => state.setCountryOfResidence)
  const clearStore = useOnboardingStore((state) => state.clear)

  useEffect(() => {
    setIsButtonEnabled(selectedCountry !== null)
  }, [selectedCountry])

  const { countries, callGetCountries, isLoading: areCountriesLoading } = useGetCountries()
  const { callCreateUser, isLoading: isCreateUserLoading } = useCreateUser()

  useEffect(() => {
    callGetCountries()
  }, [callGetCountries])

  const handleNextPress = async () => {
    const result = await callCreateUser()
    showSuccessDialog(result)
  }

  const onSuccessDialogConfirmed = () => {
    clearStore()
    const resetAction = CommonActions.reset({
      index: 0,
      routes: [
        {
          name: Routes.LoginScreen,
        },
      ],
    })

    navigation.dispatch(resetAction)
  }

  return (
    <ScreenTemplate>
      <Text variant='headlineMedium'>{Strings.onboarding_country_title}</Text>
      <Text variant='bodyMedium'>{Strings.onboarding_country_subtitle}</Text>

      <View style={CommonStyles.mt16}>
        <PaperSelect
          value={selectedCountry ? formatCountry(selectedCountry) : Strings.onboarding_country_placeholder}
          arrayList={mapCountriesToDropdownList(countries)}
          label={Strings.onboarding_country_label}
          onSelection={(item) => setSelectedCountry(findCountry(item, countries))}
          errorText=''
          multiEnable={false}
          selectedArrayList={mapCountriesToDropdownList(selectedCountry ? [selectedCountry] : [])}
        />
      </View>

      <Portal>
        <Dialog visible={isSuccessDialogShown} dismissable={false}>
          <Dialog.Title>{Strings.onboarding_success_title}</Dialog.Title>
          <Dialog.Content>
            <Text variant='bodyMedium'>{Strings.onboarding_success_message}</Text>
          </Dialog.Content>
          <Dialog.Actions>
            <Button onPress={onSuccessDialogConfirmed}>{Strings.ok}</Button>
          </Dialog.Actions>
        </Dialog>
      </Portal>

      <LoadingModal visible={areCountriesLoading || isCreateUserLoading} />

      <View style={CommonStyles.flex1} />

      <Button mode='contained' onPress={handleNextPress} disabled={!isButtonEnabled} testID='onboarding_country_button'>
        {Strings.button_next}
      </Button>
    </ScreenTemplate>
  )
}

function findCountry(selectedItem: SelectedItem, countries: Country[]): Country | null {
  if (selectedItem?.selectedList[0] === undefined) {
    return null
  } else {
    return countries.find((it) => it.id === selectedItem.selectedList[0]._id)!
  }
}

function mapCountriesToDropdownList(countries: Country[]) {
  return countries.map((country) => {
    return {
      _id: country.id,
      value: formatCountry(country),
    }
  })
}

const formatCountry = (country: Country) => `${country?.flag} ${country?.name}`
