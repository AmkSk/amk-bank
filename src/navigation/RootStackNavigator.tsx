import { createNativeStackNavigator } from '@react-navigation/native-stack'
import { RootStackParamList, Routes } from './navigationTypes'
import { IntroScreen } from '../screens/IntroScreen'
import { WelcomeScreen } from '../screens/WelcomeScreen'
import { LoginScreen } from '../screens/login/LoginScreen'
import { OnboardingCreateAccountScreen } from '../screens/onboarding/OnboardingCreateAccountScreen'
import { OnboardingEmailScreen } from '../screens/onboarding/OnboardingEmailScreen'
import { OnboardingPersonalInfoScreen } from '../screens/onboarding/OnboardingPersonalInfoScreen'
import { OnboardingCountryScreen } from '../screens/onboarding/OnboardingCountryScreen'
import { TosModalScreen } from '../screens/TosModalScreen'
import { PrivacyPolicyModalScreen } from '../screens/PrivacyPolicyModalScreen'
import { useContext } from 'react'
import { UserContext } from '../hooks/userContext'
import { TabNavigator } from './TabNavigator'

interface Props {
  shouldHideIntroScreens: boolean
}

const RootStack = createNativeStackNavigator<RootStackParamList>()

export function RootStackNavigator({ shouldHideIntroScreens }: Props) {
  const userContext = useContext(UserContext)

  return (
    <RootStack.Navigator>
      <RootStack.Group screenOptions={{ animation: 'slide_from_right', animationDuration: 100 }}>
        {shouldHideIntroScreens ? undefined : (
          <RootStack.Screen name={Routes.IntroScreen} component={IntroScreen} options={{ headerShown: false }} />
        )}
        <RootStack.Screen name={Routes.WelcomeScreen} component={WelcomeScreen} options={{ title: '' }} />
        <RootStack.Screen name={Routes.LoginScreen} component={LoginScreen} options={{ title: '' }} />
        <RootStack.Screen
          name={Routes.OnboardingCreateAccountScreen}
          component={OnboardingCreateAccountScreen}
          options={{ title: '' }}
        />
        <RootStack.Screen
          name={Routes.OnboardingEmailScreen}
          component={OnboardingEmailScreen}
          options={{ title: '' }}
        />
        <RootStack.Screen
          name={Routes.OnboardingPersonalInfoScreen}
          component={OnboardingPersonalInfoScreen}
          options={{ title: '' }}
        />
        <RootStack.Screen
          name={Routes.OnboardingCountryScreen}
          component={OnboardingCountryScreen}
          options={{ title: '' }}
        />
      </RootStack.Group>

      {userContext.userLoggedIn && (
        <RootStack.Screen name={Routes.TabNavigator} component={TabNavigator} options={{ headerShown: false }} />
      )}

      <RootStack.Group screenOptions={{ presentation: 'modal' }}>
        <RootStack.Screen name={Routes.TosModal} component={TosModalScreen} options={{ title: '' }} />
        <RootStack.Screen
          name={Routes.PrivacyPolicyModal}
          component={PrivacyPolicyModalScreen}
          options={{ title: '' }}
        />
      </RootStack.Group>
    </RootStack.Navigator>
  )
}
