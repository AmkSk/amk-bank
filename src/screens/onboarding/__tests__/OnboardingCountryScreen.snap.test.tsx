import React from 'react'
import { NativeStackScreenProps } from 'react-native-screens/native-stack'
import { RootStackParamList, Routes } from '../../../navigation/navigationTypes'
import { Provider as PaperProvider } from 'react-native-paper'
import { render } from '@testing-library/react-native'
import { OnboardingCountryScreen } from '../OnboardingCountryScreen'

const props = {} as NativeStackScreenProps<RootStackParamList, Routes.OnboardingCountryScreen>

// axios needs to be mocked, otherwise tests fail
jest.mock('axios', () => ({
  create: jest.fn(() => ({
    get: jest.fn(),
  })),
}))

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')

describe('OnboardingScreen', () => {
  it('renders correctly', () => {
    const tree = render(
      <PaperProvider>
        <OnboardingCountryScreen {...props} />
      </PaperProvider>,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
