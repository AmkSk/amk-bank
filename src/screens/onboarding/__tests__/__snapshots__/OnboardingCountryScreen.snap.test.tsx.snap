// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnboardingScreen renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTSafeAreaView
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "rgba(255, 251, 254, 1)",
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "margin": 16,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontSize": 28,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 36,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "writingDirection": "ltr",
              },
              undefined,
            ]
          }
        >
          Country of residence
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "400",
                "letterSpacing": 0.25,
                "lineHeight": 20,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "writingDirection": "ltr",
              },
              undefined,
            ]
          }
        >
          Please add the country where your permanent address is located
        </Text>
        <View
          style={
            Object {
              "marginTop": 16,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginBottom": 10,
                  "width": "100%",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "color": "#000",
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 6,
                    },
                    Object {
                      "backgroundColor": "#fff",
                      "borderColor": "black",
                      "borderRadius": 4,
                      "borderWidth": 1,
                    },
                    undefined,
                  ]
                }
                testID="text-input-outline"
              />
              <View>
                <View
                  style={
                    Array [
                      Object {
                        "paddingBottom": 0,
                      },
                      Object {
                        "minHeight": 56,
                        "paddingTop": 8,
                      },
                    ]
                  }
                >
                  <View
                    collapsable={false}
                    pointerEvents="none"
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "opacity": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "transform": Array [
                          Object {
                            "translateX": 3,
                          },
                        ],
                        "zIndex": 3,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      pointerEvents="none"
                      style={
                        Object {
                          "backgroundColor": "#fff",
                          "bottom": 4,
                          "left": 10,
                          "maxHeight": 2,
                          "opacity": 1,
                          "position": "absolute",
                          "right": 0,
                          "top": 6,
                          "transform": Array [
                            Object {
                              "translateX": -3,
                            },
                          ],
                          "width": 12,
                        }
                      }
                    />
                    <Text
                      collapsable={false}
                      maxFontSizeMultiplier={1.5}
                      numberOfLines={1}
                      style={
                        Object {
                          "backgroundColor": "#fff",
                          "color": "transparent",
                          "fontFamily": "System",
                          "fontSize": 16,
                          "fontWeight": undefined,
                          "left": 8,
                          "letterSpacing": 0.15,
                          "lineHeight": undefined,
                          "opacity": 1,
                          "paddingHorizontal": 0,
                          "position": "absolute",
                          "textAlign": "left",
                          "top": 33,
                          "transform": Array [
                            Object {
                              "translateX": 0,
                            },
                            Object {
                              "translateY": -26,
                            },
                            Object {
                              "scale": 0.75,
                            },
                            Object {
                              "scaleY": 0.2,
                            },
                          ],
                          "width": -16,
                          "writingDirection": "ltr",
                        }
                      }
                    >
                      Country
                    </Text>
                    <Text
                      collapsable={false}
                      maxFontSizeMultiplier={1.5}
                      numberOfLines={1}
                      onLayout={[Function]}
                      style={
                        Object {
                          "color": "black",
                          "fontFamily": "System",
                          "fontSize": 16,
                          "fontWeight": undefined,
                          "left": 0,
                          "letterSpacing": 0.15,
                          "lineHeight": undefined,
                          "opacity": 0,
                          "paddingHorizontal": 16,
                          "position": "absolute",
                          "textAlign": "left",
                          "top": 32,
                          "transform": Array [
                            Object {
                              "translateX": 0,
                            },
                            Object {
                              "translateY": -26,
                            },
                            Object {
                              "scale": 0.75,
                            },
                          ],
                          "writingDirection": "ltr",
                        }
                      }
                      testID="text-input-outlined-label-active"
                    >
                      Country
                    </Text>
                    <Text
                      collapsable={false}
                      maxFontSizeMultiplier={1.5}
                      numberOfLines={1}
                      style={
                        Object {
                          "color": "rgba(73, 69, 79, 1)",
                          "fontFamily": "System",
                          "fontSize": 16,
                          "fontWeight": undefined,
                          "left": 0,
                          "letterSpacing": 0.15,
                          "lineHeight": undefined,
                          "opacity": 0,
                          "paddingHorizontal": 16,
                          "position": "absolute",
                          "textAlign": "left",
                          "top": 32,
                          "transform": Array [
                            Object {
                              "translateX": 0,
                            },
                            Object {
                              "translateY": -26,
                            },
                            Object {
                              "scale": 0.75,
                            },
                          ],
                          "writingDirection": "ltr",
                        }
                      }
                      testID="text-input-outlined-label-inactive"
                    >
                      Country
                    </Text>
                  </View>
                  <TextInput
                    activeUnderlineColor="black"
                    editable={true}
                    maxFontSizeMultiplier={1.5}
                    multiline={false}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder=" "
                    placeholderTextColor="rgba(73, 69, 79, 1)"
                    selectionColor="black"
                    showSoftInputOnFocus={false}
                    style={
                      Array [
                        Object {
                          "margin": 0,
                          "zIndex": 1,
                        },
                        Object {
                          "height": 48,
                        },
                        Object {
                          "paddingBottom": 0,
                          "paddingTop": 0,
                        },
                        Object {
                          "color": "rgba(28, 27, 31, 1)",
                          "fontFamily": "System",
                          "fontSize": 16,
                          "fontWeight": undefined,
                          "letterSpacing": 0.15,
                          "lineHeight": undefined,
                          "paddingHorizontal": 16,
                          "textAlign": "left",
                          "textAlignVertical": "center",
                        },
                        false,
                        Object {
                          "marginRight": 56,
                          "paddingRight": 0,
                        },
                        undefined,
                      ]
                    }
                    testID="text-input-outlined"
                    underlineColorAndroid="transparent"
                    value="Pick your country"
                  />
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "height": 24,
                        "justifyContent": "center",
                        "position": "absolute",
                        "width": 24,
                      },
                      Object {
                        "right": 16,
                        "top": 19,
                      },
                    ]
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "alignSelf": "center",
                        "bottom": 0,
                        "end": undefined,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "shadowColor": "#000",
                        "shadowOffset": Object {
                          "height": 0,
                          "width": 0,
                        },
                        "shadowOpacity": 0,
                        "shadowRadius": 0,
                        "start": undefined,
                        "top": 0,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "shadowColor": "#000",
                          "shadowOffset": Object {
                            "height": 0,
                            "width": 0,
                          },
                          "shadowOpacity": 0,
                          "shadowRadius": 0,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          Object {
                            "alignContent": "center",
                            "alignItems": "center",
                            "backgroundColor": undefined,
                            "borderColor": "rgba(121, 116, 126, 1)",
                            "borderRadius": 15.5,
                            "borderWidth": 0,
                            "elevation": 0,
                            "height": 31,
                            "justifyContent": "center",
                            "margin": 6,
                            "overflow": "hidden",
                            "width": 31,
                          }
                        }
                      >
                        <View
                          accessibilityComponentType="button"
                          accessibilityRole="button"
                          accessibilityState={
                            Object {
                              "disabled": false,
                            }
                          }
                          accessibilityTraits="button"
                          accessible={true}
                          centered={true}
                          collapsable={false}
                          focusable={true}
                          hitSlop={
                            Object {
                              "bottom": 6,
                              "left": 6,
                              "right": 6,
                              "top": 6,
                            }
                          }
                          name="chevron-down"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Array [
                              Object {
                                "overflow": "hidden",
                              },
                              false,
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "flexGrow": 1,
                                  "justifyContent": "center",
                                },
                                Object {
                                  "borderRadius": 15.5,
                                },
                              ],
                            ]
                          }
                          testID="right-icon-adornment"
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View />
        </View>
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        />
        <View
          collapsable={false}
          style={
            Object {
              "alignSelf": undefined,
              "bottom": undefined,
              "end": undefined,
              "left": undefined,
              "position": undefined,
              "right": undefined,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
              "start": undefined,
              "top": undefined,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "rgba(28, 27, 31, 0.12)",
                  "borderColor": "transparent",
                  "borderRadius": 20,
                  "borderStyle": "solid",
                  "borderWidth": 0,
                  "minWidth": 64,
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "disabled": true,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    false,
                    Object {
                      "borderRadius": 20,
                    },
                  ]
                }
                testID="onboarding_country_button"
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "rgba(28, 27, 31, 1)",
                          "fontFamily": "System",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "lineHeight": 20,
                        },
                        Object {
                          "textAlign": "left",
                        },
                        Object {
                          "writingDirection": "ltr",
                        },
                        Array [
                          Object {
                            "marginHorizontal": 16,
                            "marginVertical": 9,
                            "textAlign": "center",
                          },
                          false,
                          Object {
                            "marginHorizontal": 24,
                            "marginVertical": 10,
                          },
                          undefined,
                          false,
                          Object {
                            "color": "rgba(28, 27, 31, 0.38)",
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          undefined,
                        ],
                      ]
                    }
                  >
                    Next
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      accessibilityLiveRegion="polite"
      accessibilityViewIsModal={true}
      collapsable={false}
      onAccessibilityEscape={[Function]}
      pointerEvents="auto"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      testID="modal"
    >
      <View
        accessibilityLabel="Close modal"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        importantForAccessibility="no"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(50, 47, 55, 0.4)",
            "flex": 1,
            "opacity": 0,
          }
        }
        testID="modal-backdrop"
      />
      <View
        pointerEvents="box-none"
        style={
          Array [
            Object {
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "marginBottom": 0,
              "marginTop": 0,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
        testID="modal-wrapper"
      >
        <View
          collapsable={false}
          style={
            Object {
              "alignSelf": undefined,
              "bottom": undefined,
              "end": undefined,
              "left": undefined,
              "position": undefined,
              "right": undefined,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.15,
              "shadowRadius": 3,
              "start": undefined,
              "top": undefined,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 1,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                  "opacity": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "rgba(255, 251, 254, 1)",
                    "borderRadius": 32,
                    "paddingHorizontal": 64,
                    "paddingVertical": 32,
                  }
                }
              >
                <View
                  accessibilityRole="progressbar"
                  accessibilityState={
                    Object {
                      "busy": true,
                    }
                  }
                  accessible={true}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "height": 24,
                        "opacity": 1,
                        "width": 24,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "alignItems": "center",
                          "bottom": 0,
                          "justifyContent": "center",
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          Object {
                            "height": 24,
                            "transform": Array [
                              Object {
                                "rotate": "45deg",
                              },
                            ],
                            "width": 24,
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          style={
                            Object {
                              "height": 12,
                              "overflow": "hidden",
                              "width": 24,
                            }
                          }
                        >
                          <View
                            collapsable={false}
                            style={
                              Object {
                                "height": 24,
                                "transform": Array [
                                  Object {
                                    "translateY": 0,
                                  },
                                  Object {
                                    "rotate": "-165deg",
                                  },
                                ],
                                "width": 24,
                              }
                            }
                          >
                            <View
                              collapsable={false}
                              style={
                                Object {
                                  "height": 12,
                                  "overflow": "hidden",
                                  "width": 24,
                                }
                              }
                            >
                              <View
                                collapsable={false}
                                style={
                                  Object {
                                    "borderColor": "rgba(103, 80, 164, 1)",
                                    "borderRadius": 12,
                                    "borderWidth": 2.4,
                                    "height": 24,
                                    "width": 24,
                                  }
                                }
                              />
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "alignItems": "center",
                          "bottom": 0,
                          "justifyContent": "center",
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          Object {
                            "height": 24,
                            "transform": Array [
                              Object {
                                "rotate": "45deg",
                              },
                            ],
                            "width": 24,
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          style={
                            Object {
                              "height": 12,
                              "overflow": "hidden",
                              "top": 12,
                              "width": 24,
                            }
                          }
                        >
                          <View
                            collapsable={false}
                            style={
                              Object {
                                "height": 24,
                                "transform": Array [
                                  Object {
                                    "translateY": -12,
                                  },
                                  Object {
                                    "rotate": "345deg",
                                  },
                                ],
                                "width": 24,
                              }
                            }
                          >
                            <View
                              collapsable={false}
                              style={
                                Object {
                                  "height": 12,
                                  "overflow": "hidden",
                                  "width": 24,
                                }
                              }
                            >
                              <View
                                collapsable={false}
                                style={
                                  Object {
                                    "borderColor": "rgba(103, 80, 164, 1)",
                                    "borderRadius": 12,
                                    "borderWidth": 2.4,
                                    "height": 24,
                                    "width": 24,
                                  }
                                }
                              />
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
</RNCSafeAreaProvider>
`;
